### PR TITLE
Add forms

### PR DESCRIPTION
**Pull request by ankitt26**
In this PR, I did - 

-  Create a method `current_user` in ApplicationController to make current user data available to all controllers.
    -  It will return the first user from the database.
- Create forms to perform the following functions:
    - Creates a Post on behalf of the `current_user` (use the method that you created in your ApplicationController).
    - Create a comment on behalf of the `current_user` (use the method that you created in your ApplicationController).
- Allow Users to add likes to Posts.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -6,5 +6,26 @@ def index
...
+  end
+
+  def create
+    @post = current_user.posts.build(post_params)
+    if @post.save
+      redirect_to user_posts_path(current_user), notice: 'Post created successfully!'
+    else
+      render :new
+    end
+  end
```
Comment (anagudelogu) [inline app/controllers/posts_controller.rb:24]: - [ ] When I try to create a `Post`, I'm getting the following errors:

image

Please consider fixing them! 🤓 🔧 

💡 _Hint: You could run a migration to set the default values of these counters to 0, so they become optional parameters._

**Review by anagudelogu — CHANGES_REQUESTED** [review]
# STATUS:  CHANGES REQUIRED  ♻️ 🚧 

Hi @ankitt26, 
I'm Andres.


While you have done a great job working on this milestone 💻 , there are some issues that you still need to work on to go to the next project. ♻️ 

You are almost there, keep up the good work! 🚀 



## Highlights
✅ Great use of Git flow. 💯 
✅ Your project is documented professionally. 📜 
✅ Descriptive PR title, summary and commit messages. 💬 

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (ankitt26) [reply app/controllers/posts_controller.rb:24]: Thanks, I will fix that.

Comment (KDlamini) [reply app/controllers/posts_controller.rb:24]: Well resolved ✔️ 

Posts, Comments, and Likes can be added succesfully. 🚀 

image

**Review by KDlamini — APPROVED** [review]
### Hi @ankitt26 🙋‍♀️,  

Excellent work! 💯 
You have done a great job adding forms to the Rails Blog app. 🎊 🥳 
Your feature is complete! There is nothing else to say other than... it's time to merge it. :shipit: 
Congratulations!!!🎊 🥳 

## Highlights 🌞

✔️  Professional pull request and README
✔️  Forms in the views that accept input for creating posts and comments implemented
✔️  Controller actions for creating new posts, comments, and likes present
✔️  Respective routes for the new actions added

# Status: Approved ✔️ 

giphy-tumblr
 
Cheers 🥂  and Happy coding!!! 👯 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @KDlamini  in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ankitt26** [comment]
Thanks for Approving the PR 😊🎉.

**Pull request merged**